gems/aws-sdk-core/lib/aws-sdk-core/instance_profile_credentials.rb

```diff
@@ -4,353 +4,217 @@
-
-    # @api private
-    class TokenRetrivalError < RuntimeError; end
-
-    # @api private
-    class TokenExpiredError < RuntimeError; end
-
-    # These are the errors we trap when attempting to talk to the
-    # instance metadata service.  Any of these imply the service
-    # is not present, no responding or some other non-recoverable
-    # error.
-    # @api private
-    NETWORK_ERRORS = [
-      Errno::EHOSTUNREACH,
-      Errno::ECONNREFUSED,
-      Errno::EHOSTDOWN,
-      Errno::ENETUNREACH,
-      SocketError,
-      Timeout::Error,
-      Non200Response
-    ].freeze
+    class InvalidProfile < RuntimeError
+      def initialize(*_args)
+        super('invalid profile name - unable to find metadata path')
+      end
+    end
 
-    # Path base for GET request for profile and credentials
+    # Legacy path base for GET request for profile and credentials
     # @api private
-    METADATA_PATH_BASE = '/latest/meta-data/iam/security-credentials/'.freeze
+    METADATA_LEGACY_PATH = '/latest/meta-data/iam/security-credentials/'
 
-    # Path for PUT request for token
+    # Extended path base for GET request for profile and credentials
     # @api private
-    METADATA_TOKEN_PATH = '/latest/api/token'.freeze
+    METADATA_EXTENDED_PATH = '/latest/meta-data/iam/security-credentials-extended/'
 
     # @param [Hash] options
-    # @option options [Integer] :retries (1) Number of times to retry
-    #   when retrieving credentials.
-    # @option options [String] :endpoint ('http://169.254.169.254') The IMDS
-    #   endpoint. This option has precedence over the :endpoint_mode.
-    # @option options [String] :endpoint_mode ('IPv4') The endpoint mode for
-    #   the instance metadata service. This is either 'IPv4' ('169.254.169.254')
-    #   or 'IPv6' ('[fd00:ec2::254]').
-    # @option options [Boolean] :disable_imds_v1 (false) Disable the use of the
-    #  legacy EC2 Metadata Service v1.
-    # @option options [String] :ip_address ('169.254.169.254') Deprecated. Use
-    #   :endpoint instead. The IP address for the endpoint.
-    # @option options [Integer] :port (80)
-    # @option options [Float] :http_open_timeout (1)
-    # @option options [Float] :http_read_timeout (1)
-    # @option options [Numeric, Proc] :delay By default, failures are retried
-    #   with exponential back-off, i.e. `sleep(1.2 ** num_failures)`. You can
-    #   pass a number of seconds to sleep between failed attempts, or
-    #   a Proc that accepts the number of failures.
-    # @option options [IO] :http_debug_output (nil) HTTP wire
-    #   traces are sent to this object.  You can specify something
-    #   like $stdout.
-    # @option options [Integer] :token_ttl Time-to-Live in seconds for EC2
-    #   Metadata Token used for fetching Metadata Profile Credentials, defaults
-    #   to 21600 seconds
-    # @option options [Callable] before_refresh Proc called before
+    # @option options [Aws::EC2Metadata] :ec2_metadata A custom EC2 metadata
```

Comment:
TODO: I'll double check these options yardoc inheritance on monday (i'm gonna check to make sure that things are rendering correctly...)